client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -190,48 +211,43 @@ def delete_prompt(self, _):
 		dialog.add_button(Gtk.STOCK_OK, Gtk.ResponseType.ACCEPT)
 		response_id = dialog.run()
 		dialog.destroy()
-		if response_id == -3:
+		if response_id == Gtk.ResponseType.ACCEPT:
 			self.delete(model, treeiter)
 
 class LocalDirectory(DirectoryBase):
-	def __init__(self, treeview, logger, upload_button, *args, **kwargs):
-		super(LocalDirectory, self).__init__(treeview, logger)
+	treeview_name = 'treeview_local'
+	def __init__(self, builder, logger, upload_button):
+		_treeview = builder.get_object('SFTPClientGUI.' + self.treeview_name)
+		super(LocalDirectory, self).__init__(_treeview, logger)
 		self.upload_button = upload_button
-		self.load_dirs(os.path.abspath(os.sep))
+		init_dir = os.getcwd()
+		self.load_dirs(init_dir)
 		self.treeview_local.set_model(self._tvmodel)
 
-	def load_dirs(self, path, parent=None):
-		counter = 0
+	def _check_perm(self, fullname):
+		r_permissions = os.access(fullname, os.R_OK)
+		w_permissions = os.access(fullname, os.W_OK)
+		x_permissions = os.access(fullname, os.X_OK)
+		perm = '   '
+		perm = perm + 'r' if r_permissions else perm + '-'
+		perm = perm + 'w' if w_permissions else perm + '-'
+		perm = perm + 'x' if x_permissions else perm + '-'
+		return perm
+
+	def _yield_dir_list(self, path):
+		names = []
 		for name in os.listdir(path):
-			if self.local_hidden and name.startswith('.'):
-				continue
-			fullname = os.path.join(path, name)
-			r_permissions = os.access(fullname, os.R_OK)
-			w_permissions = os.access(fullname, os.W_OK)
-			x_permissions = os.access(fullname, os.X_OK)
-			perm = '   '
-			perm = perm + 'r' if r_permissions else perm + '-'
-			perm = perm + 'w' if w_permissions else perm + '-'
-			perm = perm + 'x' if x_permissions else perm + '-'
-			date_modified = '   ' + time.strftime("%y-%m-%d %H:%M:%S", time.localtime(os.path.getmtime(fullname)))
-			isFolder = stat.S_ISDIR(os.stat(fullname).st_mode)
-			if isFolder:
-				icon = Gtk.IconTheme.get_default().load_icon('folder', 20, 0)
-				if not (r_permissions and x_permissions):
-					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
-				current = self._tvmodel.append(parent, (name, icon, fullname, perm, None, -1, date_modified))
-			else:
-				file_size = os.path.getsize(fullname)
-				hr_file_size = '   ' + boltons.strutils.bytes2human(file_size)
-				if not (r_permissions and x_permissions):
-					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
-				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
-				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
-			if isFolder and (r_permissions and x_permissions):
-				self._tvmodel.append(current, [None, None, None, None, None, None, None])
-			counter += 1
-		if counter < 1:
-			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+			names.append(name)
```

Comment:
Instead of keeping a list in memory this should use Python's `yield` keyword. It will be more efficient that way.